scripts/circle_deploy.sh

```diff
@@ -64,3 +56,21 @@ else
  echo "...publishing"
  npm publish
 fi
+
+#
+# gh-pages
+#
+echo "...deploying gh-pages"
+git push -f $CIRCLE_BRANCH:gh-pages
+git checkout gh-pages
+gulp docs
+git add .
+git add -f docs/build
+
+if [[ -n $(git status --porcelain) ]]; then
+  echo "...starting push, gh-pages is dirty after build"
+  git commit -n -m "deploy gh-pages by $CIRCLE_USERNAME [ci skip]"
+  git push origin gh-pages
+else
+  echo "...skipping push, gh-pages is clean after build"
+fi
```

Comment:
Pulled the gh-pages bit into it's own block here. Haven't tested this yet (first deploy will).  It is pretty straight forward.

1. Push the current branch to gh-pages (deploy is master only right now)
1. Checkout gh-pages
1. Build docs, add everything, force add the ignored `docs/build`

    >This is ok because we're on the gh-pages branch here, master won't start tracking it.  Removes need for an integration branch.
1. If we have local changes, commit and push them